### ISSUE TITLE
npm@2.12.1 onward - git+https is broken

### ISSUE THREAD

**Issue by timbru31** [npm/npm#8881]
Hey there,

it seems that the changes
e7f56e5a97fcf1c52d5c5bee71303b0126129815 and the potential fix 94ca4a711619ba8e40ce3d20bc42b13cdb7611b7
do not fix the issue with git+https

npm@2.12.0 works for me while npm@2.12.1 doesn't

It's failing with fatal: ``Unable to find remote helper for 'git+https'``

npm-debug.log is here: http://codeshare.io/UV75C

dependecy is used like this
```js
"dependencies": {
    "normalize.less": "^1.0.0",
    "patternplate": "git+ssh://git@git.sinnerschrader.com/patternplate/patternplate.git#v0.11.7",
    "react": "^0.13.3"
  }
```
Thanks for looking into this, if any information is missing, I'll try to provide it! :)
Cheers,
Tim

**Comment by othiym23** [comment]
I think those are probably Node version and not npm versions – have you tried upgrading to `npm@latest` (which is currently `npm@2.12.1`), and if so, are you still seeing this problem?

**Comment by timbru31** [comment]
Sorry mixed node and npm versions. 9 hours of working and no coffee...

I've updated the description.
Node is at v0.12.6 and the issue **only** happens with npm@2.12.1

**Comment by zkat** [comment]
Hey! So I took a look, and it seems you're using an invalid `git+https://` url as one of the dependencies of `patternplate`. This probably worked in the past because of how things used to be normalized, but that was part of the thing that was fixed recently. It sounds like we should still try and support these invalid URLs (at least with a warning) since we did used to accept them, intentionally or not.

Anyway, for the sake of unblocking you:

* `git+ssh://` accepts `scp`-style syntax location specs now, and should be the only one that does (that is, things like `git@yourserver.io:relative/path` (note the `:`)
* `git+https://` only accepts _valid URIs_ right now, which means the relative path syntax above will not work. You'll need to either change those to `git+https://` or path directly to them with `git+https://git@yourserver.io/absolute/path/to/project` (see `/` instead of `:` after host).

Leaving this open so we can add something like a warning for folks who were relying on the previous behavior.
